kasa/smart/modules/firmware.py

```diff
@@ -98,11 +100,19 @@ def update_available(self):
 
     async def get_update_state(self):
         """Return update state."""
-        return await self.call("get_fw_download_state")
+        if self._device._is_cloud_connected:
+            return await self.call("get_fw_download_state")
```

Comment:
Inverse would be nicer for these conditionals, i.e., fail early. That is, if we don't want to allow executing these _even_ when we think there is no cloud connectivity, leaving the decision to the user? Just food for thought.

Reply:
Done

Reply:
Done